### PR TITLE
Update SSRC API client

### PR DESCRIPTION
**Pull request by erikeldridge**
Internal client updates.

This is based on PR #2456, which defined the API.

## Discussion

Working with @lahirumaramba and @trekforever.

## Testing

Ran npm test and all tests pass.

### CODE REVIEW

File: src/remote-config/remote-config-api-client-internal.ts [hunk 1]
```diff
@@ -21,10 +21,16 @@ import { PrefixedFirebaseError } from '../utils/error';
...
+  ListVersionsOptions,
+  ListVersionsResult,
+  RemoteConfigTemplate,
+  RemoteConfigServerTemplateData
+} from './remote-config-api';
 
 // Remote Config backend constants
-const FIREBASE_REMOTE_CONFIG_V1_API = 'https://firebaseremoteconfig.googleapis.com/v1';
+// Honors env param to enable URL override independent of binary change.
+const FIREBASE_REMOTE_CONFIG_URL_BASE = process.env.FIREBASE_REMOTE_CONFIG_URL_BASE || 'https://firebaseremoteconfig.googleapis.com';
```
Comment (erikeldridge) [inline src/remote-config/remote-config-api-client-internal.ts:33]: @lahirumaramba I've found it helpful here and in the Web SDK to be able to point the SDK at a local server for testing while in development, and issue reproduction after release. I don't see other references to `process.env` in this SDK, though. Do you have a preferred way to reference an env var?

Comment (lahirumaramba) [reply src/remote-config/remote-config-api-client-internal.ts:33]: Thanks! This makes sense to me. We use `process.env` to handle other env variables https://github.com/firebase/firebase-admin-node/blob/5fe41189279c0a037db3fb48298a1adc657e9701/src/storage/storage.ts#L48

Let's make a note to document this new env var in our public docs

File: src/remote-config/remote-config-api.ts [hunk 2]
```diff
@@ -54,6 +54,27 @@ export interface RemoteConfigCondition {
   tagColor?: TagColor;
 }
 
+/**
+ * Interface representing a Remote Config condition in the data-plane.
+ * A condition targets a specific group of users. A list of these conditions make up
+ * part of a Remote Config template.
+ */
```
Comment (lahirumaramba) [inline src/remote-config/remote-config-api.ts:61]: Let's get TW reviews on these. Thanks!

**Review by lahirumaramba — APPROVED** [review]
LGTM! Thanks!

File: src/remote-config/remote-config-api.ts [hunk 3]
```diff
@@ -54,6 +54,27 @@ export interface RemoteConfigCondition {
   tagColor?: TagColor;
 }
 
+/**
+ * Interface representing a Remote Config condition in the data-plane.
```
Comment (jenh) [inline src/remote-config/remote-config-api.ts:58]:
```suggestion
 * Represents a Remote Config condition in the dataplane.
```

File: src/remote-config/remote-config-api.ts [hunk 4]
```diff
@@ -135,7 +156,7 @@ export interface RemoteConfigParameterGroup {
 }
 
 /**
- * Interface representing a Remote Config template.
+ * Interface representing a Remote Config client template.
```
Comment (jenh) [inline src/remote-config/remote-config-api.ts:159]:
```suggestion
 * Represents a Remote Config client template.
```

File: src/remote-config/remote-config-api.ts (hunk 2 above)
Comment (erikeldridge) [reply src/remote-config/remote-config-api.ts:61]: Ah. Good point. I was focused on this being an internal client and overlooked that it updated the API. I've added @jenh as a reviewer.

File: src/remote-config/remote-config-api.ts [hunk 5]
```diff
@@ -167,6 +188,58 @@ export interface RemoteConfigTemplate {
   version?: Version;
 }
 
+/**
+ * Interface representing the data in a Remote Config server template.
```
Comment (jenh) [inline src/remote-config/remote-config-api.ts:192]:
```suggestion
 * Represents the data in a Remote Config server template.
```

File: src/remote-config/remote-config-api.ts [hunk 6]
```diff
@@ -167,6 +188,58 @@ export interface RemoteConfigTemplate {
...
+   */
+  conditions: RemoteConfigServerCondition[];
+
+  /**
+   * Map of parameter keys to their optional default values and optional conditional values.
+   */
+  parameters: { [key: string]: RemoteConfigParameter };
+
+  /**
+   * ETag of the current Remote Config template (readonly).
```
Comment (jenh) [inline src/remote-config/remote-config-api.ts:206]:
```suggestion
   * Current Remote Config template ETag (read-only).
```

File: src/remote-config/remote-config-api.ts [hunk 7]
```diff
@@ -167,6 +188,58 @@ export interface RemoteConfigTemplate {
...
+  readonly etag: string;
+
+  /**
+   * Version information for the current Remote Config template.
+   */
+  version?: Version;
+}
+
+/**
+ * Interface representing a stateful abstraction for a Remote Config server template.
```
Comment (jenh) [inline src/remote-config/remote-config-api.ts:217]:
```suggestion
 * Represents a stateful abstraction for a Remote Config server template.
```

File: src/remote-config/remote-config-api.ts [hunk 8]
```diff
@@ -167,6 +188,58 @@ export interface RemoteConfigTemplate {
...
+ */
+export interface RemoteConfigServerTemplate {
+
+  /**
+   * Cached {@link RemoteConfigServerTemplateData}
```
Comment (jenh) [inline src/remote-config/remote-config-api.ts:222]:
```suggestion
   * Cached {@link RemoteConfigServerTemplateData}.
```

File: src/remote-config/remote-config-api.ts [hunk 9]
```diff
@@ -167,6 +188,58 @@ export interface RemoteConfigTemplate {
...
+   */
+  cache: RemoteConfigServerTemplateData;
+
+  /**
+   * A {@link RemoteConfigServerConfig} containing default values for Config
```
Comment (jenh) [inline src/remote-config/remote-config-api.ts:227]:
```suggestion
   * A {@link RemoteConfigServerConfig} that contains default Config values.
```

File: src/remote-config/remote-config-api.ts [hunk 10]
```diff
@@ -167,6 +188,58 @@ export interface RemoteConfigTemplate {
...
+   */
+  defaultConfig: RemoteConfigServerConfig;
+
+  /**
+   * Evaluates the current template to produce a {@link RemoteConfigServerConfig}
```
Comment (jenh) [inline src/remote-config/remote-config-api.ts:232]:
```suggestion
   * Evaluates the current template to produce a {@link RemoteConfigServerConfig}.
```

File: src/remote-config/remote-config-api.ts [hunk 11]
```diff
@@ -167,6 +188,58 @@ export interface RemoteConfigTemplate {
...
+   */
+  evaluate(): RemoteConfigServerConfig;
+
+  /**
+   * Fetches and caches the current active version of the
+   * {@link RemoteConfigServerTemplate} of the project.
```
Comment (jenh) [inline src/remote-config/remote-config-api.ts:238]:
```suggestion
   * project's {@link RemoteConfigServerTemplate}.
```

File: src/remote-config/remote-config-api.ts [hunk 12]
```diff
@@ -289,3 +362,8 @@ export interface ListVersionsOptions {
    */
   endTime?: Date | string;
 }
+
+/**
+ * Type representing the configuration produced by evaluating a server template.
```
Comment (jenh) [inline src/remote-config/remote-config-api.ts:367]:
```suggestion
 * Represents the configuration produced by evaluating a server template.
```

File: src/remote-config/remote-config-api-client-internal.ts [hunk 13]
```diff
@@ -21,10 +21,16 @@ import { PrefixedFirebaseError } from '../utils/error';
...
+import {
+  ListVersionsOptions,
+  ListVersionsResult,
+  RemoteConfigTemplate,
+  RemoteConfigServerTemplateData
+} from './remote-config-api';
 
 // Remote Config backend constants
-const FIREBASE_REMOTE_CONFIG_V1_API = 'https://firebaseremoteconfig.googleapis.com/v1';
+// Honors env param to enable URL override independent of binary change.
```
Comment (jenh) [inline src/remote-config/remote-config-api-client-internal.ts:32]:
```suggestion
/**
  * Allows the `FIREBASE_REMOTE_CONFIG_URL_BASE` environment
  * variable to override the default API endpoint URL.
  */
```

**Review by jenh — CHANGES_REQUESTED** [review]
LGTM with minor changes.

File: src/remote-config/remote-config-api-client-internal.ts (hunk 1 above)
Comment (erikeldridge) [reply src/remote-config/remote-config-api-client-internal.ts:33]: Added a row to our burndown to document this env var (go/rc-ai-burndown).

**Comment by erikeldridge** [comment]
Thank you for reviewing, @lahirumaramba, @jenh!

@jenh, I've updated the PR with your suggested changes and re-requested review from you.

**Pull request merged**